modules/tinymce/src/core/main/ts/SelectionOverrides.ts

```diff
@@ -192,38 +124,28 @@ const SelectionOverrides = (editor: Editor): SelectionOverrides => {
         return;
       }
 
-      const contentEditableRoot = getContentEditableRoot(editor, targetElm);
-      if (contentEditableRoot) {
-        if (isContentEditableFalse(contentEditableRoot)) {
+      // Remove needs to be called here since the mousedown might alter the selection without calling selection.setRng
+      // and therefore not fire the AfterSetSelectionRange event.
+      removeElementSelection();
+      hideFakeCaret();
+
+      const closestContentEditable = getContentEditableRoot(editor, targetElm);
+      if (isContentEditableFalse(closestContentEditable)) {
+        e.preventDefault();
+        FakeCaretUtils.selectNode(editor, closestContentEditable).each(setElementSelection);
+      } else {
+        ClosestCaretCandidate.closestFakeCaretCandidate(rootNode, e.clientX, e.clientY).each((caretInfo) => {
           e.preventDefault();
-          FakeCaretUtils.selectNode(editor, contentEditableRoot).each(setElementSelection);
-        } else {
-          removeElementSelection();
-
-          // Check that we're not attempting a shift + click select within a contenteditable='true' element
-          if (!(isContentEditableTrue(contentEditableRoot) && e.shiftKey) && !RangePoint.isXYWithinRange(e.clientX, e.clientY, selection.getRng())) {
-            hideFakeCaret();
-            selection.placeCaretAt(e.clientX, e.clientY);
-          }
-        }
-      } else if (isFakeSelectionTargetElement(targetElm)) {
-        FakeCaretUtils.selectNode(editor, targetElm).each(setElementSelection);
-      } else if (isFakeCaretTarget(targetElm) === false) {
-        // Remove needs to be called here since the mousedown might alter the selection without calling selection.setRng
-        // and therefore not fire the AfterSetSelectionRange event.
-        removeElementSelection();
-        hideFakeCaret();
+          const range = showCaret(1, caretInfo.node as HTMLElement, caretInfo.position === ClosestCaretCandidate.FakeCaretPosition.Before, false);
```

Comment:
Not something to do now, but we probably should use an enum here instead of `1` to mean forwards one day 🤷